### PR TITLE
Purge cache when new version is pushed or a version is yanked

### PR DESCRIPTION
**Pull request by sonalkr132**
closes: #1344 

Don't we need to purge memcache and Fastly when a version is yanked?

### CODE REVIEW

**Comment by dwradcliffe** [comment]
Yeah you're right, we need both.

File: lib/fastly.rb [hunk 1]
```diff
@@ -30,4 +33,10 @@ def self.purge_key(key, soft = false)
     Rails.logger.debug "Fastly purge url=#{url} status=#{json['status']} id=#{json['id']}"
     json
   end
+
+  def self.purge_api_cdn(gem_name)
+    ["deps/v1/#{gem_name}", "info/#{gem_name}", "versions", "names"].each do |path|
```
Comment (segiddins) [inline lib/fastly.rb]: isn't `"deps/v1/#{gem_name}"` only stored in memcached?

Comment (dwradcliffe) [inline lib/fastly.rb]: Good catch

Comment (sonalkr132) [inline lib/fastly.rb]: How do we know what is stored on fastly cdn?

Comment (dwradcliffe) [inline lib/fastly.rb]: At this point we should assume everything passes through Fastly. For `"deps/v1/#{gem_name}"`, it is not an actual endpoint, it's just a memcached key.

**Pull request merged**